lm75bd/lm75bd.c

```diff
@@ -26,7 +26,17 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
+  uint8_t pointerRegister = 0;
+  i2cSendTo(LM75BD_OBC_I2C_ADDR, &pointerRegister, 1);
+
+  uint8_t buf[2];
+  i2cReceiveFrom(LM75BD_OBC_I2C_ADDR, buf, 2);
+
+  if (!(buf[0] & 128)) {
+    *temp = 0.125 * (buf[0] << 3 | buf[1] >> 5);
+  } else {
+    *temp = -0.125 * (((buf[0] << 3 | buf[1] >> 5) ^ ((1 << 11) - 1)) + 1);
+  }
```

Comment:
This code is way to complicated it can be simplified to `int16_t temperature=((buf[0]<<8|buff[1])>>5)*0.125` and `*temp=temperature`